inference-engine/tests/functional/plugin/cpu/single_layer_tests/batch_to_space.cpp

```diff
@@ -0,0 +1,126 @@
+
+#include <shared_test_classes/single_layer/batch_to_space.hpp>
+#include "ngraph_functions/builders.hpp"
+#include "test_utils/cpu_test_utils.hpp"
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ngraph::opset3;
+
+namespace CPULayerTestsDefinitions  {
+
+typedef std::tuple<
+        LayerTestsDefinitions::batchToSpaceParamsTuple,
+        CPUSpecificParams> BatchToSpaceLayerCPUTestParamSet;
+
+class BatchToSpaceCPULayerTest : public testing::WithParamInterface<BatchToSpaceLayerCPUTestParamSet>,
+        virtual public LayerTestsUtils::LayerTestsCommon, public CPUTestsBase {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<BatchToSpaceLayerCPUTestParamSet> &obj) {
+        LayerTestsDefinitions::batchToSpaceParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+        std::ostringstream result;
+        result << LayerTestsDefinitions::BatchToSpaceLayerTest::getTestCaseName(
+                testing::TestParamInfo<LayerTestsDefinitions::batchToSpaceParamsTuple>(basicParamsSet, 0));
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        LayerTestsDefinitions::batchToSpaceParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        std::vector<size_t> inputShape;
+        std::vector<int64_t> blockShape, cropsBegin, cropsEnd;
+        InferenceEngine::Precision netPrecision;
+        std::tie(blockShape, cropsBegin, cropsEnd, inputShape, netPrecision, inPrc, outPrc, inLayout, outLayout, targetDevice) = basicParamsSet;
+        inPrc = outPrc = netPrecision;
+        selectedType = std::string("unknown_") + netPrecision.name();
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+        auto paramOuts = ngraph::helpers::convert2OutputVector(
+                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+        auto b2s = ngraph::builder::makeBatchToSpace(paramOuts[0], ngPrc, blockShape, cropsBegin, cropsEnd);
+        b2s->get_rt_info() = getCPUInfo();
+        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(b2s)};
+        function = std::make_shared<ngraph::Function>(results, params, "BatchToSpace");
+    }
+};
+
+TEST_P(BatchToSpaceCPULayerTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "BatchToSpace");
+};
+
+namespace {
+    const std::vector<Precision> precisions = {
+            Precision::I8,
+            Precision::FP32,
+            Precision::BF16
+    };
+    const std::vector<std::vector<int64_t>> blockShape4D  = {{1, 1, 2, 2}};
```

Comment:
Can you add more test data? I think that one block shape and one crops end is not enough